### PR TITLE
Build and run Metal tests in CI

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86062
* #86061

Fixes https://github.com/pytorch/pytorch/issues/84172

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: .jenkins/pytorch/macos-build.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ fi
 
 cross_compile_arm64() {
   # Cross compilation for arm64
-  USE_DISTRIBUTED=1 CMAKE_OSX_ARCHITECTURES=arm64 MACOSX_DEPLOYMENT_TARGET=11.0 USE_MKLDNN=OFF USE_QNNPACK=OFF WERROR=1 BUILD_TEST=OFF python setup.py bdist_wheel
+  USE_DISTRIBUTED=1 CMAKE_OSX_ARCHITECTURES=arm64 MACOSX_DEPLOYMENT_TARGET=11.0 USE_MKLDNN=OFF USE_QNNPACK=OFF WERROR=1 BUILD_TEST=OFF USE_PYTORCH_METAL=1 python setup.py bdist_wheel
```
Comment (kimishpatel) [inline .jenkins/pytorch/macos-build.sh:38]: Yeah this is what @malfet alluded to in the sev.

**Review by kimishpatel — APPROVED** [review]

**Review by malfet — APPROVED** [review]
I have stashed change that fixes sign compare warnings in metal code, but this will work for the time being

**Comment by malfet** [comment]
Here is the fix for sign-unsigned compare: https://github.com/pytorch/pytorch/pull/86068

**Pull request closed without merging**